src/electron/ipc/ipc-renderer-shim.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Action } from 'common/flux/action';
+import { IpcRenderer } from 'electron';
+import { SetSizePayload } from 'electron/flux/action/window-frame-actions-payloads';
+import {
+    IPC_APP_VERSION_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_CLOSE_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_ENTERFULLSCREEN_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_MAXIMIZE_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_MINIMIZE_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_RESTORE_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_SETSIZEANDCENTER_CHANNEL_NAME,
+    IPC_BROWSERWINDOW_UNMAXIMIZE_CHANNEL_NAME,
+    IPC_MAIN_WINDOW_INITIALIZED_CHANNEL_NAME,
+} from 'electron/ipc/ipc-channel-names';
+
+export class IpcRendererShim {
+    public constructor(private readonly ipcRenderer: IpcRenderer) {}
+
+    private readonly invokeScope: string = 'IpcRendererShim';
+
+    public initialize(): void {
+        this.ipcRenderer.on(IPC_BROWSERWINDOW_MAXIMIZE_CHANNEL_NAME, this.onMaximize);
```

Comment:
We have two types of events. 
1.) Event 1 - renderer process sends request to main process to maximize.
2.) Event 2 - Main process invokes window.maximize. After window is maximized, main process gets notified & it calls window.webContents.send to notify renderer process that maximize event has occured.

Currently, in this PR both Event 1 & Event 2, shares the same name. Event though they are used for different puposes. (1st event to submit maximize request & second event to notify maximize event has occured.) So, may be better to have different names to avoid confusion.

Reply:
Done. Channels originating from the renderer start IPC_FROMRENDERER. Channels originating from the browserWindow start IPC_FROMBROWSERWINDOW. Also updated the private methods to be more specific in their names